### PR TITLE
Reduce the blocking time of the write lock when deleting space

### PR DESCRIPTION
**Pull request by flymysql**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 

#5755

#### Description:

When the number of parts of a space is relatively large and the amount of data written is also large, it will block for a long time in the removeSpace function when deleting the space. Affect business read and write

## How do you solve it?

Shorten the range of holding locks

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [x] Function test
- [x] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by wey-gu** [comment]
Wow, thanks @flymysql for the great contribution.

**Review by critical27 — APPROVED** [review]
LGTM~

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -516,36 +516,43 @@ std::shared_ptr<Part> NebulaStore::newPart(GraphSpaceID spaceId,
...
       // before calling removeSpace, meta client would call removePart to remove all parts in
       // meta cache, which do not contain learners, so we remove them here
       if (part->isLearner()) {
         removePart(spaceId, partId, false);
       }
     }
-    auto& engines = spaceIt->second->engines_;
+    this->spaces_.erase(spaceIt);
+  }
+  if (ok(spaceOr)) {
```
Comment (SuperYoko) [inline src/kvstore/NebulaStore.cpp:539]: See  line 527 ?  Remove redundent if sentance~

**Review by SuperYoko — APPROVED** [review]

File: src/kvstore/NebulaStore.cpp (hunk 1 above)
Comment (flymysql) [reply src/kvstore/NebulaStore.cpp:539]: This is not redundant. A separate if branch is used to shorten the scope of the lock.

Comment (SuperYoko) [reply src/kvstore/NebulaStore.cpp:539]: I am just afraid that someone will refactor this.  maybe use 
```c++
  { lock gurad; code;} 
  {other code;}
```
And it's okay now.

**Comment by wey-gu** [comment]
Cheers! @critical27 @flymysql @SuperYoko

**Review by critical27 — APPROVED** [review]

**Pull request merged**